calour/ratio_experiment.py

```diff
@@ -0,0 +1,254 @@
+        values are the database specific data for the experiment (i.e. annotations for dbbact)
+
+    See Also
+    --------
+    Experiment
+    '''
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def heatmap(self, *args, **kwargs):
+        '''Plot a heatmap for the ratio experiment.
+
+        This method accepts exactly the same parameters as input with
+        its parent class method and does exactly the sample plotting.
+
+        The only difference is that by default, it uses the diverging 'coolwarm' colormap and bad_color set to white.
+        You can always set it to other colormap/bad_color as explained in :meth:`.Experiment.heatmap`.
+
+        See Also
+        --------
+        Experiment.heatmap
+
+        '''
+        if 'cmap' not in kwargs:
+            kwargs['cmap'] = 'coolwarm'
+        if 'bad_color' not in kwargs:
+            kwargs['bad_color'] = 'white'
+        if 'clim' not in kwargs:
+            min_val = np.abs(np.nanmin(self.data))
+            max_val = np.abs(np.nanmax(self.data))
+            clim_val = np.max([max_val, min_val])
+            kwargs['clim'] = (-clim_val, clim_val)
+        super().heatmap(*args, **kwargs)
+
+    @classmethod
+    def from_exp(self, exp, common_field, group_field, value1, value2=None, threshold=5):
+        '''Create a RatioExperiment from two groups of samples in an experiment
+
+        Parameters
+        ----------
+        exp: calour.Experiment
+            The experiment to take the 2 groups of samples from.
+        common_field: str
+            Name of the sample metadata field to calculate the ratio within.
+            For example, to calculate the ratio between 2 timepoints for each individual,
+            common_field with be 'subject_id'
+        group_field: str
+            Name of the sample metadata field on which to divide to 2 groups for ratio calculation.
+            For example, to calculate the ratio between 2 timepoints for each individual,
+            group_field with be 'time_point'
+        value1 : str or iterable of str
+            Values for field that will be assigned to the first (nominator) sample group
+        value2: str or iterable of str or None, optional
+            If not None, values for field that will be assigned to the second (denominator) sample group.
+            If None, use all samples not in the first sample group as the second sample group.
+        threshold: float or None, optional
+            If not None, assign each data value<threshold to threshold. If both nominator and denominator are < threshold,
+            the resulting ratio will assigned be np.nan.
+            For example, in an AmpliconExperiment, ratios calculated between low read numbers are not reliable. So it is advised to set threshold to approx. 10
+
+        Returns
+        -------
+        RatioExperiment
+        The samples are all the unique values of common_field
```

Comment:
I think it's not all unique values in common_field, only unique values which contain value1 and value2 in group_field?

Reply:
cool. fixed.